core/dbt/contracts/graph/model_config.py

```diff
@@ -446,6 +446,7 @@ class NodeConfig(NodeAndTestConfig):
         default_factory=Docs,
         metadata=MergeBehavior.Update.meta(),
     )
+    constraints_enabled: Optional[bool] = False
```

Comment:
I'm open to changing it after community feedback! There are exceptions that I flow through into `DatabaseErrors` vs. `ParsingErrors` because we get error handling for free(think: data type validation, number of columns mismatch SQL). 

This does NOT validate order or name consistency across the SQL statement to your point. This is a much larger effort for `static typing` for `dbt`.

Reply:
_nit_: If it has a default value `False` then it doesn't need to be an `Option type`, it could just be `bool`.

It's a shame the python typing lib uses the word `Optional` as a keyword; it's the default value that makes providing the attribute _optional_ when instantiating the class.  

On another note, a bool with a default value of `None` forces the use of `Optional[bool]` which only makes sense if you must have a lenient (public) API or if there's inheritance compatibility issues.

Reply:
@davehowell Good catch! It looks like @gshank already had the same thought, and implemented this change over in #7002, along with the config rename :)